Warning message in building WiringPi on Raspberry 4

When issuing a build I get the following warning:
drcNet.c: In function \u2018authenticate\u2019:
drcNet.c:112:24: warning: \u2018%s\u2019 directive writing up to 1023 bytes into a region of size 1021 [-Wformat-overflow=]
drcNet.c:89:14:
       return &buf [10] ;
              ~~~~~~~~~  
drcNet.c:112:24:
   sprintf (salted, "$6$%s$", challenge) ;
                        ^~
drcNet.c:112:3: note: \u2018sprintf\u2019 output between 5 and 1028 bytes into a destination of size 1024
   sprintf (salted, "$6$%s$", challenge) ;

--- a/wiringPi/drcNet.c
+++ b/wiringPi/drcNet.c
@@ -76,12 +76,12 @@ static int remoteReadline (int fd, char *buf, int max)
 
 static char *getChallenge (int fd)
 {
-  static char buf [1024] ;
+  static char buf [512] ;
   int num ;
 
   for (;;)
   {
-    if ((num = remoteReadline (fd, buf, 1023)) < 0)
+    if ((num = remoteReadline (fd, buf, 511)) < 0)
       return NULL ;
     buf [num] = 0 ;
 

